### PR TITLE
Story/bods 1797

### PR DESCRIPTION
**Pull request by amitkpmguk**
PR includes following ticket
https://kpmgengineering.atlassian.net/browse/BODS-1797

### CODE REVIEW

File: transit_odp/otc/weca/registry.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+from transit_odp.otc.models import Service
+from transit_odp.otc.weca.client import APIResponse, WecaClient
+
+logger = getLogger(__name__)
+
+
+class Registry:
+    def __init__(self) -> None:
+        self._client = WecaClient()
+        self.services = []
```
Comment (ankitmahajan1) [inline transit_odp/otc/weca/registry.py:14]: services is of type Dataframe

File: transit_odp/otc/weca/loaders.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        if len(self.registry.services) > 0:
+            self.delete_services()
+            self.load_services()
+
+        logger.info("WECA job finished the execution")
+
+    def load_services(self):
+        logger.info("Loading services into the database")
+        service_objects = []
+        self.registry.services.reset_index()
```
Comment (ankitmahajan1) [inline transit_odp/otc/weca/loaders.py:25]: need to pass inplace = true for persisting the  reseting index

File: transit_odp/otc/weca/registry.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    def process_services(self) -> None:
+        """
+        Fetch records from the weca api, and implement following
+        1. Remove duplicate services and service name combination
+        2. Merge the services based on service name seprated by pipe
+        3. Ignore the services which belongs to OTC
+        """
+        self.fetch_all_records()
+        self.convert_to_pandas()
```
Comment (ankitmahajan1) [inline transit_odp/otc/weca/registry.py:82]: better function name would be convert_services_to_df

File: transit_odp/otc/weca/registry.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        logger.info("Removing WECA duplicates.")
+        self.remove_duplicates()
+        logger.info(
+            "Merging WECA records for same registration numbers by making service name seprated from pipe."
+        )
+        self.merge_service_numbers()
```
Comment (ankitmahajan1) [inline transit_odp/otc/weca/registry.py:88]: need to add check if self.services is empty df

File: transit_odp/otc/weca/registry.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        """
+        To ignore the services which belongs to OTC but are comming in WECA API
+        Expectation is such scenarios will not happen but if they are those will be
+        skipped from the dataframe
+        """
+        registrations_queryset = Service.objects.filter(
+            registration_number__in=self.services["registration_number"].tolist(),
+            api_type__isnull=True,
+        ).values_list("registration_number")
+        registrations_to_ignore = [service[0] for service in registrations_queryset]
```
Comment (ankitmahajan1) [inline transit_odp/otc/weca/registry.py:60]: better to rename service to registration

File: transit_odp/otc/factories.py [hunk 6]
```diff
@@ -12,11 +14,11 @@
     TrafficAreas,
 )
 from transit_odp.otc.dataclasses import Licence, Operator, Registration, Service
-from transit_odp.otc.models import Licence as LicenceModel, UILta
+from transit_odp.otc.models import Licence as LicenceModel
 from transit_odp.otc.models import LocalAuthority
 from transit_odp.otc.models import Operator as OperatorModel
 from transit_odp.otc.models import Service as ServiceModel
-from factory.django import DjangoModelFactory
+from transit_odp.otc.models import UILta
```
[the remark this answers is not in the document]
Comment (amitkpmguk) [reply transit_odp/otc/factories.py:21]: As discussed, no need to change

File: transit_odp/otc/weca/registry.py (hunk 4 above)
Comment (amitkpmguk) [reply transit_odp/otc/weca/registry.py]: Updated

File: transit_odp/otc/weca/registry.py (hunk 3 above)
Comment (amitkpmguk) [reply transit_odp/otc/weca/registry.py]: Updated

File: transit_odp/otc/weca/registry.py (hunk 5 above)
Comment (amitkpmguk) [reply transit_odp/otc/weca/registry.py]: Updated variable names to service

File: transit_odp/otc/weca/registry.py (hunk 1 above)
Comment (amitkpmguk) [reply transit_odp/otc/weca/registry.py]: Updated

File: transit_odp/otc/weca/loaders.py (hunk 2 above)
Comment (amitkpmguk) [reply transit_odp/otc/weca/loaders.py]: Updated

**Review by ankitmahajan1 — DISMISSED** [review]
approved

**Review by ankitmahajan1 — DISMISSED** [review]
approved

**Review by ankitmahajan1 — APPROVED** [review]
approved

**Pull request merged**